includes/sanitizers/class-amp-base-sanitizer.php

```diff
@@ -342,4 +345,108 @@ public function remove_invalid_attribute( $element, $attribute, $args = array()
 			}
 		}
 	}
+
+	/**
+	 * Get data-amp-* values from the parent node 'figure' added by editor block.
+	 *
+	 * @param DOMNode $node Base node.
+	 * @return array AMP data array.
+	 */
+	public function get_data_amp_attributes( $node ) {
+		$attributes = array();
+
+		// Editor blocks add 'figure' as the parent node for images. If this node has data-amp-layout then we should add this as the layout attribute.
+		$parent_node = $node->parentNode;
+		if ( 'figure' === $parent_node->tagName ) {
+			$parent_attributes = AMP_DOM_Utils::get_node_attributes_as_assoc_array( $parent_node );
+			if ( isset( $parent_attributes['data-amp-layout'] ) ) {
+				$attributes['layout'] = $parent_attributes['data-amp-layout'];
+			}
+			if ( isset( $parent_attributes['data-amp-noloading'] ) && true === filter_var( $parent_attributes['data-amp-noloading'], FILTER_VALIDATE_BOOLEAN ) ) {
+				$attributes['noloading'] = $parent_attributes['data-amp-noloading'];
+			}
+		}
+
+		return $attributes;
+	}
+
+	/**
+	 * Set AMP attributes.
+	 *
+	 * @param array $attributes Array of attributes.
+	 * @param array $amp_data Array of AMP attributes.
+	 * @return array
+	 */
+	public function set_data_amp_attributes( $attributes, $amp_data ) {
+		if ( isset( $amp_data['layout'] ) ) {
+			$attributes['data-amp-layout'] = $amp_data['layout'];
+		}
+		if ( isset( $amp_data['noloading'] ) ) {
+			$attributes['data-amp-noloading'] = '';
+		}
+		return $attributes;
+	}
+
+	/**
+	 * Set attributes to node's parent element according to layout.
+	 *
+	 * @param DOMNode $node Node.
+	 * @param array   $new_attributes Attributes array.
+	 * @param string  $layout Layout.
+	 * @return array New attributes.
+	 */
+	public function set_attachment_layout_attributes( $node, $new_attributes, $layout ) {
```

Comment:
Same as above:

> I think instead of `set_` this should be prefixed with `filter_`, since it's not actually setting any attributes on a node. That would align with `filter_attributes` method in the image sanitizer.

But that gets tricky here because here there are side-effects on the parent node. So it's not so clear what is the best to do.

Reply:
Changed this to `filter_` within dbc1c7c since even with setting the parent node attributes it seems to be more consistent. One option could be to separate setting the parent node attributes from the filtering method into a separate method, would that be a more clear / better alternative? Thoughts?